src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -2199,6 +2201,7 @@ public int[] executeBatch() throws SQLServerException, BatchUpdateException, SQL
             loggerExternal.finer(toString() + ACTIVITY_ID + ActivityCorrelator.getCurrent().toString());
         }
         checkClosed();
+        connection.unprepareUnreferencedPreparedStatementHandles(false);
```

Comment:
there are more occurrences of executeStatement in the file should we be doing this for every one of them?